WcaOnRails/app/controllers/incidents_controller.rb

```diff
@@ -11,13 +11,24 @@ class IncidentsController < ApplicationController
   ]
 
   def index
-    base_model = Incident.includes(:competitions, :incident_tags)
-    if current_user&.can_view_delegate_matters?
-      @incidents = base_model.all
-    else
-      @incidents = base_model.resolved
+    respond_to do |format|
+      format.html {}
+      format.json do
+        @incidents = Incident.all
+        # // todo: filter/search?
+        total_entries = @incidents.length
+        @incidents = @incidents
+                     .page(params[:page] || 1)
+                     .per(params[:entries_per_page] || 10)
```

Comment:
Rails has a `paginate` helper that automatically unwraps and applies these values for you, cf. `controllers/api/v0/competitions_controller.rb#index`

(that method also has some rudimentary search functionality, through the `Competitions.search` invocation. Might be a good first orientation for a place to start your own incidents search)

Reply:
Should I switch to using that? It seems more confusing to me, since everything happens implicitly and I can't find a documentation page clearly laying everything out for `paginate`. Plus I'm not sure how to get values from Response Headers.

I had several attempts trying to copy various existing searches and none of them worked. Philippe agreed to help me out with it (on Slack).

Reply:
Hm, I'm surprised to see that `Posts` is using the same approach. I just found out that the Rails helper is not really a Rails helper but rather a 3rd party integration provided by https://github.com/davidcelis/api-pagination. Sorry, my bad.

Anyways, I just wanted to point out that it's something worth investigating. The library that I linked above references an IETF RFC which I believe is a good practice to follow. Then again we shouldn't tear our legs out just to follow some arbitrary RFCs if there's a much simpler implementation that makes life easier for you.

Really just brainstorming ideas at this point!

Reply:
Also: @viroulep Merci mille fois pour le soutien supplémentaire <3

Reply:
Fun fact: both your approaches use the same thing :')

The `per` and `page` options come from a third party gem, kaminari.
api-pagination auto detects the paginator, and should call kaminari too, but expose the page/per-page/total in the response headers, not in the payload.

If my statement above is correct, `paginate :collection, per_page: 10` should fire the same sql query (with the appropriate LIMIT clause) as `collection.per(10).page(params[:page])`, because `paginate` takes into account the `params[:page]` and calls kaminari appropriately.

Posts probably don't use paginate because I didn't have a good idea to forward the request headers to the underlying components at that time (or didn't want to bother with that just yet), but there is probably a good way to do that (for instance by having a `headers` field exposed by the hook in addition to the `data` one).

Reply:
This has been updated to use `paginate`.